### ISSUE TITLE
Error when refresh token expires

### ISSUE THREAD

**Issue by hgoscenski** [jfarmer08/wyze-api#9]
After token expires, it appears that without checking the code/msg of the response the `wyze-api` is not attempting to refresh the token.

```
[2/13/2024, 3:08:58 PM] [Wyze Smart home] Refreshing devices...
[2/13/2024, 3:08:58 PM] [Wyze Smart home] Performing request: app/v2/home_page/get_object_list
[2/13/2024, 3:08:58 PM] [Wyze Smart home] Request config: {"method":"POST","url":"app/v2/home_page/get_object_list","data":{"access_token":"<SNIP>","app_name":"com.hualai.WyzeCam","app_ver":"wyze_developer_api","app_version":"wyze_developer_api","phone_id":"wyze_developer_api","phone_system_type":"1","sc":"wyze_developer_api","sv":"wyze_developer_api","ts":1707836938928},"baseURL":"https://api.wyzecam.com"}
[2/13/2024, 3:08:59 PM] [Wyze Smart home] API response PerformRequest: {"code":"2001","msg":"access token is error","traceId":"265862178700f9a907ae2dc979f8daa7","data":{}}
[2/13/2024, 3:08:59 PM] [Wyze Smart home] access token is error
[2/13/2024, 3:08:59 PM] [Wyze Smart home] Error getting devices: TypeError: Cannot read properties of undefined (reading 'length')
```

**Comment by jfarmer08** [comment]
So you think we need to add that back in? I wish I knew all the response returns. 
@tnoor-co Could you help any with that?

**Comment by hgoscenski** [comment]
@jfarmer08 I do, my instance was unable to control any accessories and was getting the above message. 

The way we are handling recovery from the login failure was via catching the exception, it appears that the Wyze API does not return a failing HTTP code, but rather a 200 OK with a msg/code inside of it. 

Looking at the logs I believe that `code == 1` indicates success.

**Comment by jfarmer08** [comment]
I pushed version 1.1.4

**Comment by jfarmer08** [comment]
@hgoscenski Do you mind if I direct message you?

**Comment by hgoscenski** [comment]
Not at all, please go ahead, harrison at hgoscenski.com.

**Comment by jfarmer08** [comment]
@hgoscenski  looks like result.data.code is null every time. Not sure what is the best way to fix. this.

**Comment by hgoscenski** [comment]
Interesting, we could potentially do this:

```
if (result.data.code && result.data.code != 1)
```

PR: https://github.com/jfarmer08/wyze-api/pull/11

**Comment by jfarmer08** [comment]
That does work.

**Comment by tj-noor** [comment]
> @tnoor-co Could you help any with that?

Hey @jfarmer08 - sorry I was a bit swamped today. I did get some time to work on it and was able to gather some error codes and messages we could match on, and then went down a rabbit hole. I initially hoped Wyze was using JWT's, so we could attempt to extract the expiration from it, but the tokens were encrypted - so y'alls current solution of estimating is the most optimal one. I then came across rate limiting headers and added some handling there along with retrying the requests selectively. There were a decent amount of changes made on it that may not be fully on topic here - so apologies in advance for it. Happy to collaborate on it further if you're open to it. https://github.com/jfarmer08/wyze-api/pull/12
